Source/Gral/GRALShapeFiles/Shapereader.cs

```diff
@@ -152,21 +152,27 @@ public IEnumerable<object> ReadShapeFile(string filename)
 					line.Parts = new int[line.NumParts];
 					line.NumPoints = readIntLittle(data, recordContentStart + 40);
 					line.Points = new GralDomain.PointD[line.NumPoints];
+					line.PointsZ = new double[line.NumPoints];
+
 					int partStart = recordContentStart + 44;
 					for (int i = 0; i < line.NumParts; i++)
 					{
 						line.Parts[i] = readIntLittle(data, partStart + i * 4);
 					}
-					int pointStart = recordContentStart + 44 + 4 * line.NumParts;
 
-					for (int i = 0; i < line.NumPoints; i++)
+                    int pointStart = recordContentStart + 44 + 4 * line.NumParts;
+                    for (int i = 0; i < line.NumPoints; i++)
 					{
 						line.Points[i].X = readDoubleLittle(data, pointStart + (i * 16));
-						line.Points[i].Y = readDoubleLittle(data, pointStart + (i * 16) + 8);
+						int index = pointStart + (i * 16) + 8;
+						line.Points[i].Y = readDoubleLittle(data, index);
+						index = pointStart + 16 * line.NumPoints + 16 + 8 * i;
```

Comment:
This needs to be checked at further tests. And we do not support the optional data for max and min heights.